Add new devices, and overhaul Xbox One implementation

FranticRain opened the pull request:
Apparently Rock Candy put out a new version of their controller with a different product ID, so here it is.

HVSNick commented:
Hey all! Just bumping this since we noticed PDP changes the model number of the Rock Candy controller and a new build will be needed to use it.

FranticRain commented:
Added the Xbox One Elite controller to this pull request. I don't know if this one will have a hard time with the Travis.yml again. That thing seems busted if it can't handle changes to the Info.plist.

d235j commented:
@FranticRain I shut it off because it was being a pain. I'll see if I can get it working later.

HVSNick commented:
Dumb guy here but does this mean it's ready to go? :)


- N -

*Nick Ahrens*
Creative Producer | Hidden Variable Studios
*Social Bits! @robotpants*

On Tue, Oct 27, 2015 at 10:13 AM, David Ryskalczyk <notifications@github.com
> wrote:

> @FranticRain <https://github.com/FranticRain> I shut it off because it
> was being a pain. I'll see if I can get it working later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/360Controller/360Controller/pull/151#issuecomment-151573781>
> .
>

TyanColte commented:
Can somebody commit this pull request so i can get a release and test this?

herrbischoff commented:
@TyanColte: Merging this pull request would not automatically result in a new release. Your best bet for now would be to compile it yourself with Yosemite and Xcode 6.

herrbischoff commented:
@TyanColte: I just had a couple of minutes and compiled the pull request version complete with code signing. The only thing I couldn't get to work was the package creation, so you will have to move the files to their places manually. If you trust me, please find the download below.

Download: https://herrbischoff.com/360ControllerPR151.zip

TyanColte commented:
@herrbischoff What did you compile this with, i keep getting a message that the extension can't be used. I'm using OS X el capitan version 10.11.1 any help would be appreciated. Thanks

herrbischoff commented:
@TyanColte: Sorry for this. It should be signed now.

TyanColte commented:
@herrbischoff Thanks, I'll give it a go.

TyanColte commented:
@herrbischoff got it installed, it gave me the same error at first, but then i rebooted and it stopped giving me that error. However it's still not recognizing the elite controller. Nor is it recognizing my regular first generation Xbone controller with the latest firmware.

TyanColte commented:
@FranticRain @herrbischoff if it makes a difference i'm looking at the system profiler on my MBP 2010 with OS X El Capitan and it's saying the Product ID is 0x02e3 and the Vendor ID is 0x045e where in the kext info.plist it says 739 for product and 1118 for vendor.

FranticRain commented:
Yeah, I can confirm. `kextutil` is giving me signing errors.
```
Hnir:360ControllerPR151 drewmills$ sudo kextutil 360Controller.kext
Notice: 360Controller.kext has debug properties set.
Diagnostics for 360Controller.kext:
Warnings: 
    Personality CFBundleIdentifier differs from containing kext's (not necessarily a mistake, but rarely done): 
        ChatPadKeyboardEvents

Code Signing Failure: code signature is invalid
Warnings: 
    Personality CFBundleIdentifier differs from containing kext's (not necessarily a mistake, but rarely done): 
        ChatPadKeyboardEvents

Untrusted kexts are not allowed
ERROR: invalid signature for com.mice.driver.Xbox360Controller, will not load
```

@TyanColte When we get this signing business figured out, there is one more step you need to do. In the `/Library/Extensions` folder, you'll need to type `sudo chown -R root:wheel 360Controller.kext`. You may also need to do this for the alias in `/System/Library/Extensions`, but I don't remember.

TyanColte commented:
I would do this myself but i don't have 99 bucks i can spend on a membership to the developer program. So there's that. I have a friend who has done this so he may be able to help me out.

FranticRain commented:
Does your friend specifically have a kext signing certificate though? Its a process where you have to send a request to Apple with your reason for needing a kext signing cert that happens after you get a developer license. It usually takes months to get it accepted, if they decide that you gave a good enough reason. You have to fill out [this form](https://developer.apple.com/contact/kext/) to apply.

TyanColte commented:
@FranticRain Oh well maybe not then. Well crap, why does apple have to make these things so dang hard. I guess we'll keep going on our current course.

TyanColte commented:
I wonder how many security measures I'd have to bypass to allow my macbook to run unsigned kexts.

FranticRain commented:
Well, I'm almost done with this, so I'm going to go ahead and put together a changelog. I'll try to iron out the final bits tomorrow. For some reason, the "Reset Rebinding" button decided to stop working, even though I swear I didn't touch that code. And I've completely ignored the wireless implementation again, I'll get that figured out.

Okay, here we go, patch notes:
 - Updated `fiddleReport`, it uses a cleaner pass by reference style now. This makes it easier to make subclasses of `Xbox360ControllerClass`
 - `_60Controller.h` now includes `ControlStruct.h`. Removed includes from `_60Controller.cpp` and `Controller.cpp`. This is so `fiddleReport` can take `XBOX360_HAT`s instead of a `void*`
 - Added `XboxOnePretend360Class` which handles Xbox One controllers pretending to be 360 controllers
 - Changed `XboxOneControllerClass` to report correct VID and PID. No longer pretends to be a 360 controller
 - Added `XboxOneEliteControllerClass` so that we can add special handling if we want for the Elite controller
 - Made new `remapButtons` and `remapAxes` for Xbox One controllers
 - Added new `xboxOnehid.h` Xbox One report descriptor
 - Fixed remapping of Xbox One controllers that aren't pretending to be 360
 - Added `MakeSettingsChanges()` to Xbox360Peripheral for changes that need to take place immediately after a settings read
 - Xbox One controller triggers technically have a range of 0-1023, but I've normalized it to 0-255 in `fixTriggers()`. I don't know if this is a limitation of the gamepad API or the preference pane
 - Added `remapButtons(UInt16* buttons, UInt8 mapping[])` in Xbox One classes to get rid of some repeated code (For some reason I couldn't get it to work with lines 722-735 of `Controller.cpp`, no idea why)
 - Added an enum so specific buttons are easier to find in the Xbox One button packet. Also added enum for the Elite paddle packet
 - Added `pramga mark`s to `Controller.cpp` to mark the start of every class to try and make jumping around in code easier

RodrigoCard commented:
Oh, Great! I will do some testing! :)

TyanColte commented:
@FranticRain Do you have an unsigned version I can test?

FranticRain commented:
@TyanColte Sure! For some reason Dropbox is down, so here's a G Drive link. I'll update it to a Dropbox link later. https://drive.google.com/open?id=0B5PLCKoMd4zGNUR1czJWUTNhcEE

TyanColte commented:
@FranticRain Thanks! I'll give it a shot and let you know how it works.

TyanColte commented:
@FranticRain it said i needed permission to access the file. I sent you a request for permission.

FranticRain commented:
@TyanColte Jeez, I'm so bad with G Drive, give the link another shot, it should be public now.

TyanColte commented:
@FranticRain worked that time. Thanks!

TyanColte commented:
@FranticRain 

Diagnostics for /Library/Extensions/360Controller.kext:
Warnings:
    Personality CFBundleIdentifier differs from containing kext's (not necessarily a mistake, but rarely done):
        ChatPadKeyboardEvents

Code Signing Failure: not code signed
Warnings:
    Personality CFBundleIdentifier differs from containing kext's (not necessarily a mistake, but rarely done):
        ChatPadKeyboardEvents

kext-dev-mode allowing invalid signature -67062 0xFFFFFFFFFFFEFA0A for kext "/Library/Extensions/360Controller.kext"
(kernel) kxld[com.mice.driver.Xbox360Controller]: The following symbols are unresolved for this kext:
(kernel) kxld[com.mice.driver.Xbox360Controller]: 	__ZN22Xbox360ControllerClassC1Ev
(kernel) kxld[com.mice.driver.Xbox360Controller]: 	__ZN22XboxOneControllerClassC1Ev
(kernel) kxld[com.mice.driver.Xbox360Controller]: 	__ZN22XboxOnePretend360ClassC1Ev
(kernel) kxld[com.mice.driver.Xbox360Controller]: 	__ZN27XboxOneEliteControllerClassC1Ev
(kernel) kxld[com.mice.driver.Xbox360Controller]: 	__ZN27XboxOriginalControllerClassC1Ev
(kernel) Can't load kext com.mice.driver.Xbox360Controller - link failed.
(kernel) Failed to load executable for kext com.mice.driver.Xbox360Controller.
(kernel) Kext com.mice.driver.Xbox360Controller failed to load (0xdc008016).
(kernel) Failed to load kext com.mice.driver.Xbox360Controller (error 0xdc008016).
Failed to load /Library/Extensions/360Controller.kext - (libkern/kext) link error.
Check library declarations for your kext with kextlibs(8).
Christophers-MacBook-Pro:extensions tyancolte$

TyanColte commented:
@FranticRain 

Christophers-MacBook-Pro:extensions tyancolte$ kextlibs 360Controller.kext
For all architectures:
    com.apple.iokit.IOHIDFamily = 2.0
    com.apple.iokit.IOUSBFamily = 900.4.1
    com.apple.kpi.iokit = 15.0
    com.apple.kpi.libkern = 15.0
    com.apple.kpi.mach = 15.0

For x86_64:
    5 symbols not found in any library kext.

FranticRain commented:
@TyanColte Something about uploading it to G Drive messed up it. I'll look into it.

FranticRain commented:
@TyanColte Ah, it was my .xcodeproj being a jerk. Problem solved! https://drive.google.com/file/d/0B5PLCKoMd4zGV0lndDQ4YVVQbmM/view

TyanColte commented:
@FranticRain Is there any way i could have you join my TeamSpeak server? I want to talk about the issues with the kext but it would take WAYYYYY too long to try and type it all out. There are some buttons that aren't in the right places. Let me know and i'll email you the server address and password.

FranticRain commented:
@TyanColte Is it that the d-pad is inverted and such? If not, just email me, and I'll see what I can do.

TyanColte commented:
@FranticRain Sent you an email with my teamspeak server details in it. I'm in the Other/Off Topic channel when you're able to get in.

FranticRain commented:
@RodrigoCard We worked it out, but it looks like we suffer from the settings bug again. His buttons were all improperly mapped until he started toggling `Swap Sticks`. Looks like the settings bug strikes again.

TyanColte commented:
@FranticRain well so much for working it out. Steam picks up that there's a controller in there, but the controls are all screwed up. And i just tried playing broforce and it sees the controller also but not only are the buttons messed up but so are the axises i don't know what's going on. i was able to remap the controls in steam to pick up the right things but broforce did not like it whatsoever. i'll test it with a few others but i think it may end up the same.

FranticRain commented:
@TyanColte That is almost undoubtedly an issue where the game is specifically looking for an Xbox 360 layout. This is exactly why I was wary of #175. I'll work on getting that feature back in.

TyanColte commented:
@FranticRain ok, sounds good.

FranticRain commented:
@RodrigoCard I'm also looking for a place to add a "Pretend to be 360" check to the Pref Pane. it would only be useful on Xbox One controllers though. Any idea what tab that should go on? Maybe on "Advanced," if we make the deadzones boxes a little shorter, we could fit it in.

FranticRain commented:
@TyanColte This is super janky, but it works with my 2013 controller. I won't be pushing these changes because I'm not happy with it, but on the main page there is a "Pretend 360" checkbox. Check it, and then try playing games. Just unzip, and put this in `/Library/PreferencePanes`, and then double click on it. https://db.tt/r0N8oaeD
If you do revert back to a normal preference pane, make sure you uncheck the checkbox first.

TyanColte commented:
@FranticRain that did not work.

FranticRain commented:
@TyanColte After you loaded checked the check box, can you reload the preference pane and see if the controller name is `Xbox One Wired Controller (Xbox 360)`? Because if that works, then it should be acting exactly like the .14 Xbox One controller.

TyanColte commented:
@FranticRain how do you mean reload the preference pane?

FranticRain commented:
@TyanColte After you check the box, just back out and then open it up again.

TyanColte commented:
@FranticRain Still says elite controller, but the test page doesn't work anymore. However the controller still works and is still lit up.

FranticRain commented:
@TyanColte Give it a shot with your 2013 controller, I'll try and figure out why it isn't working for the Elite controller.

TyanColte commented:
worked with the 2013 controller.

FranticRain commented:
@TyanColte Does Broforce play properly with that change?

TyanColte commented:
I'll check.

TyanColte commented:
@FranticRain nope.

FranticRain commented:
@TyanColte And this game works with 0.15, correct? Because I have no idea what the problem is.

TyanColte commented:
@FranticRain That’s a good question, i haven’t tried it on the standard stuff. I’ll do that and get back to you.
> On Dec 5, 2015, at 4:00 PM, Drew Mills <notifications@github.com> wrote:
> 
> @TyanColte <https://github.com/TyanColte> And this game works with 0.15, correct? Because I have no idea what the problem is.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/360Controller/360Controller/pull/151#issuecomment-162250809>.
>

FranticRain commented:
@TyanColte Before you do, unplug the controller and reconnect it after you've done "Pretend 360". It looks like there might be a minor bug that that fixes.

TyanColte commented:
@FranticRain Ok, i'll do that.

TyanColte commented:
@FranticRain with the elite or the 2013 one?

FranticRain commented:
@TyanColte 2013. I can't guarantee any of this works with the Elite controller.

TyanColte commented:
@FranticRain ok, sounds good.

TyanColte commented:
@FranticRain this is a bad game to test with on the mac, it's buggy as hell on here so i'm not going to use that as a test anymore. I'm going to try something else.

HVSNick commented:
​Was the new Rock Candy given up on? :/​


On Sat, Dec 5, 2015 at 2:07 PM, TyanColte <notifications@github.com> wrote:

> @FranticRain <https://github.com/FranticRain> this is a bad game to test
> with on the mac, it's buggy as hell on here so i'm not going to use that as
> a test anymore. I'm going to try something else.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/360Controller/360Controller/pull/151#issuecomment-162252353>
> .
>

FranticRain commented:
@HVSNick Its in the pull request. As soon as this gets released, it'll be in there.

HVSNick commented:
Rad! Thanks :)


On Mon, Dec 7, 2015 at 3:45 PM, Drew Mills <notifications@github.com> wrote:

> @HVSNick <https://github.com/HVSNick> Its in the pull request. As soon as
> this gets released, it'll be in there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/360Controller/360Controller/pull/151#issuecomment-162707131>
> .
>

TyanColte commented:
@FranticRain any luck with this? i went back to the stock version 15b3 because i wanted to use my '13 controller for some things. I remapped the controls in Broforce, and it helped a lot. However it's supposed to natively support the controller so i'm not sure what's going on.

FranticRain commented:
@TyanColte So did you test the game with the 2013 controller before reverting? How did it act? Did you tell it to pretend 360 and then unplug and replug? What was the bad mapping like? How does it compare to the mapping on the old version of the driver? You said you remapped the controls in Broforce, so the controls were still wrong even after you reverted? Did you try remapping the controls with the new version of the driver? Based on what you are telling me, it just looks like the game doesn't have good default bindings. I personally don't own Broforce, so any forward progress is going to be based fully on what you can tell me.

TyanColte commented:
@FranticRain i did test with a 2013 controller, it acted exactly as the elite controller did. I told it to pretend 360 and it didn't change anything. I think broforce may be an outlier it seemed to be pretty buggy on the mac as it was. nothing worked as expected with the controller. left stick worked some times but then other times it didn't to change menus, sometimes it would go into menus and start the game just from me moving the stick. None of the buttons worked at all until after i remapped them. I didn't try remapping them with the new version of the driver.

FranticRain commented:
@TyanColte It is a game level problem. From a quick google search, it looks like you have to rebind the controls within the game, even on an Xbox 360 controller. This is not an issue with this pull request, nor is it an issue with the driver. If you can find something that acts differently between .15 and this pull request, that's an issue.

RodrigoCard commented:
@FranticRain So, any problems with this one pull from your tests?
I really want to make a release with all this cool stuff :)

If you need some kext sign please contact me directly, I have not checked my notifications here recently.

FranticRain commented:
@RodrigoCard It looks like it works on my end. There is no "Pretend 360" toggle in the pref pane though. I could never figure out where it would best fit in, and it only applies to Xbox One. I haven't done any extensive testing thanks to finals though. We did finally get the Elite controller to stop pitching a fit though. Just remember that some people may need to toggle some values on the pref pane to get settings to update on the new version. For some reason, bindings don't trigger this update.

RodrigoCard commented:
So, the bindings does not work correctly for any controller? or just some of them?

FranticRain commented:
It works for all controllers. It just doesn't always take effect immediately, for whatever reason. For example, the Elite controller had incorrect bindings, but as soon as we toggled "swap sticks" the bindings cleared up. It continued to be correct after deactivating swap sticks. I don't quite get it myself.

larianstudios commented:
The game is going green any day now. Really, really cool if fixes will be in. 

Cheers!

FranticRain commented:
@RodrigoCard I got my new signing certificate. I was going to put together a quick build for the minor controller additions, but I couldn't figure out one step. I got it to compile using the build script, but it looks like the .pkgproj has no data set up in it, so its all messed up. How do I get that to work?

TyanColte commented:
@FranticRain YAY! i can finally re-secure my mac. Please let me know when you get a signed version of the driver out. Also I'm sorry i havn't been contributing much lately. It's been a busy holiday season.

bestwnh commented:
@FranticRain make sure you install the Package(http://s.sudre.free.fr/Software/Packages/about.html) App before you run the `build.sh`. And then run the `build.sh`. (I don't remember is the `makedmg.sh` in folder `Install360Controller` need to run after the build.) Finally you will get a `Install360Controller.mpkg` in build folder.
If you need help I can make some build and give you more clear steps. I just can't wait for the new version come out. Thanks anyway.
@RodrigoCard go go go~~~🏃🏃🏃

FranticRain commented:
@bestwnh Thanks a lot! Any idea why `packagesbuild` would be throwing an "Unable to sign the data" error? I have a `Developer ID Installer` certificate, so it should work. I can't find where to point Packages at my correct certificate though. Thanks for all the help.

FranticRain commented:
Okay, this build is so super unofficial, I'm not even going to put it on the release page. This release pretty much only helps @HVSNick and @scarmichael
This release contains the updates the updated Rock Candy Xbox One controller, and whatever nonsense that Red Octane controller is.
https://drive.google.com/file/d/0B5PLCKoMd4zGajhnQWxJUXA2UVk/view?usp=sharing

bestwnh commented:
@FranticRain I know the "Unable to sign the data". I met it before(It's so so unfriendly). It's just need a simple change: Open the packages app with the `.pkgproj` file, then at the default settings page of the project, change the `Format` from `Fat` to `Bundle`, then you will notice the certification badge at the top right corner disappear. Then you can successfully build the `.mpkg` file.

scarmichael commented:
@FranticRain Worked like a charm! Installed the latest, rebooted and both controllers are recognized now. Wonderful. A 3-year-old will be very happy to know he can now play Lego Marvel Super Heroes with his big brother.

TyanColte commented:
@FranticRain Is this unofficial release supposed to add the Elite Controller support as well?

FranticRain commented:
@TyanColte It only contains the two controllers listed. The Elite controller + Xbox One rewrite stuff still needs some work, mostly involving the prefpane.

larianstudios commented:
Game is OUT now on Steam :)

WaferMouse commented:
This seems to be where most people are talking about the next version.  My Razer Wildcat isn't working with beta3, and I can do my best helping with testing any unofficial releases.  Here's the USB dets if it helps get the ball rolling:

Product ID:	0x0a03
Vendor ID:	0x1532

FranticRain commented:
@WaferMouse In the future, please create a new issue for this kind of problem. Issues exist for this exact purpose, and you shouldn't clutter up the conversation about a pull request with an issue like this.

Either way, this should fix your problem. Let me know if it does.
https://drive.google.com/file/d/0B5PLCKoMd4zGWDBwWjk3bUtzYjg/view?usp=sharing

RodrigoCard commented:
@FranticRain @larianstudios sorry for the silence.

@FranticRain could you merge @larianstudios ' rumble fix and include in
this pull request?

I am  in a vacation (with a crappy 3g) but brought my mac and a xbox one
controller. I can make a release tomorrow, :)

Em quinta-feira, 24 de dezembro de 2015, Drew Mills <
notifications@github.com
<javascript:_e(%7B%7D,'cvml','notifications@github.com');>> escreveu:

> @WaferMouse <https://github.com/WaferMouse> In the future, please create
> a new issue for this kind of problem. Issues exist for this exact purpose,
> and you shouldn't clutter up the conversation about a pull request with an
> issue like this.
>
> Either way, this should fix your problem. Let me know if it does.
>
> https://drive.google.com/file/d/0B5PLCKoMd4zGWDBwWjk3bUtzYjg/view?usp=sharing
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/360Controller/360Controller/pull/151#issuecomment-167150710>
> .
>


-- 
Enviado do Gmail para celular

WaferMouse commented:
@FranticRain That release works flawlessly, thanks for that! Sorry about the lapse in etiquette.

FranticRain commented:
@RodrigoCard Okay, it's all up to date. The only remaining thing is a checkbox that makes Xbox One controllers pretend to be 360 controllers. It only does anything for Xbox One controllers, and probably shouldn't show up for any other controller type. The checkbox needs to hook into the "Pretend360" setting flag, and it should work perfectly. I couldn't figure out where it should go, so I'll leave that part up to you.

FranticRain commented:
Oh boy this went super sideways. I'm going to close this, and clean it all up. Give me a bit.

RodrigoCard commented:
People with elite controllers, please check the build in #184

https://github.com/360Controller/360Controller/pull/184


Em sexta-feira, 25 de dezembro de 2015, Drew Mills <notifications@github.com>
escreveu:

> Closed #151 <https://github.com/360Controller/360Controller/pull/151>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/360Controller/360Controller/pull/151#event-500383624>.
>


-- 
Enviado do Gmail para celular

jebkam commented:
Hey there, @FranticRain! I've been following closely the Rock Candy Controller Not Recognized issue @HVSNick brought up, since I recently got one of those controllers. 

The latest "super unofficial" .dmg fixed the recognition aspect and my Mac now recognizes the device as an Xbox One Wired Controller (Yay!). Although it's not detecting any of my inputs in the "Controller Test" tab. 

![screen shot 2016-01-03 at 5 53 25 pm](https://cloud.githubusercontent.com/assets/7463940/12081353/544e5408-b244-11e5-8a19-a45d6dc21016.png)

I checked on OpenEmu and it recognizes as an official Microsoft wired controller and some of the games I tested on Steam (OlliOlli & Towerfall) did also recognize the device was connected but both (OpenEmu & Steam) failed to detect my inputs.

So far, I installed the super unofficial build (and restarted the Mac). I'm not an expert and I'm not sure if I might have missed  a step or something in the comments, or if I had to do any additional manual installations, that's why I'm reaching out. Any help is appreciated! Also, Happy New Year to everyone while I'm at it. I don't know what type of sorcery you're all using to make this work, but I really appreciate that you're taking the time to make this happen.

FranticRain commented:
@jebkam Can you check your `Console.app` located in `Applications/Utilities` for any sort of error message? When you get it open, in the side right under the "FILES" header, you should have a thing called `system.log` listed. Select it. Then connect your controller. Please put any output you get from it here. Secondly, does the controller have a light on it? Does it light up for OS X like it does for Windows? And finally, if it has a removable cable, which I don't think it does, have you tried a different one? The alternate cable is the least likely fix, but sometimes it solves the problem.

jebkam commented:
@FranticRain Apologies for the delay and thanks for the quick response! To answer the questions first, Yes the controller does have the light on it and it turns on thanks to the super unofficial build. I'm not sure if it lights in OSX the same way as for Windows, though but here's how it looks to me if it helps:
![20160104_010335](https://cloud.githubusercontent.com/assets/7463940/12084479/eb132854-b281-11e5-915a-439dc0ded21d.png)
It does have a removable cable although I have no way of testing it immediately with a replacement to see if it fixes the problem. I'd have to buy one, so I might need some help too figuring out the specs for the cable I should purchase. 
 ![20160104_010720](https://cloud.githubusercontent.com/assets/7463940/12084484/faa9bf94-b281-11e5-80ca-c3241a20afef.png)
Now for the console data, I wasn't exactly sure about the errors I had to keep an eye for, but here's what happened when I plugged it in (apologies in advance if there's no useful info in there):
`Jan  4 00:22:43 Cerulean kernel[0]: start - unable to find chatpad interface`
`Jan  4 00:23:01 Cerulean syslogd[45]: ASL Sender Statistics`
`Jan  4 00:23:23 Cerulean aslmanager[134]: -[CacheDeleteServiceListener servicePeriodic:info:replyBlock:]:124 periodic callback is NULL`
`Jan  4 00:23:25 Cerulean com.apple.lakitu[388]: Checking iCDP status for DSID 1094909731 (checkWithServer=0)`
`Jan  4 00:23:26 Cerulean cdpd[1031]: Received new connection <NSXPCConnection: 0x7ff02bd29690> connection from pid 388, checking entitlements...`
`Jan  4 00:23:26 Cerulean cdpd[1031]: Saw change in network reachability (isReachable=2)`
`Jan  4 00:23:26 Cerulean cdpd[1031]: Unknown client type with bundleID 'com.apple.lakitu'`
`Jan  4 00:23:26 Cerulean cdpd[1031]: Accepting new connection <NSXPCConnection: 0x7ff02bd29690> connection from pid 388 with entitlements mask 3`
`Jan  4 00:23:26 Cerulean cdpd[1031]: KeychainGetICDPStatus: keychain: -25300`
`Jan  4 00:23:26 Cerulean cdpd[1031]: KeychainGetICDPStatus: status: off`
`Jan  4 00:23:26 Cerulean com.apple.lakitu[388]: iCDP status for DSID 1094909731 is DISABLED`
`Jan  4 00:23:28 Cerulean accountsd[273]: AIDA Notification plugin running`
`Jan  4 00:23:28 Cerulean com.apple.AddressBook.InternetAccountsBridge[391]: Checking iCDP status for DSID 1094909731 (checkWithServer=0)`
`Jan  4 00:23:28 Cerulean com.apple.AddressBook.InternetAccountsBridge[391]: Daemon connection invalidated!`
`Jan  4 00:23:28 Cerulean com.apple.AddressBook.InternetAccountsBridge[391]: XPC Error while checking if iCDP is enabled for DSID 1094909731: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.cdp.daemon was invalidated." UserInfo={NSDebugDescription=The connection to service named com.apple.cdp.daemon was invalidated.}`
`Jan  4 00:23:28 Cerulean com.apple.AddressBook.InternetAccountsBridge[391]: [AOSAccounts] : [__49-[ACAccountStore(iCloudAccount) icaAppleAccounts]_block_invoke] : accountsWithAccountType returned error: Error Domain=com.apple.accounts Code=9 "(null)"`
`Jan  4 00:23:30 Cerulean sandboxd[139] ([391]): com.apple.Addres(391) deny mach-lookup com.apple.cdp.daemon`
`Jan  4 00:24:24 Cerulean System Preferences[1040]: Failed to connect (bindingInputColumn) outlet from (Pref360ControlPref) to (NSTableColumn): missing setter or instance variable`
`Jan  4 00:24:24 Cerulean System Preferences[1040]: Failed to connect (linearDeadzoneLeft) outlet from (Pref360ControlPref) to (NSButton): missing setter or instance variable`
`Jan  4 00:24:24 Cerulean System Preferences[1040]: Failed to connect (linearDeadzoneRight) outlet from (Pref360ControlPref) to (NSButton): missing setter or instance variable`
`Jan  4 00:24:24 Cerulean System Preferences[1040]: Failed to connect (xoneRumbleOptions) outlet from (Pref360ControlPref) to (NSPopUpButton): missing setter or instance variable`
`Jan  4 00:24:24 Cerulean System Preferences[1040]: Unable to simultaneously satisfy constraints:`
`	(`
`	   "<NSLayoutConstraint:0x7fadb30c5c20 H:[MyWhole360Controller:0x7fadb30c5cf0(465)]>",`
`	    "<NSLayoutConstraint:0x7fadb30ad6d0 H:|-(NSSpace(20))-[MyWhole360Controller:0x7fadb30c5cf0]   (Names: '|':NSView:0x7fadb30adc00 )>",`
`	   "<NSAutoresizingMaskLayoutConstraint:0x7fadb0f55ba0 h=-&- v=-&- H:|-(0)-[NSView:0x7fadb30adc00]   (Names: '|':NSTabView:0x7fadb30adf30 )>",`
`	    "<NSAutoresizingMaskLayoutConstraint:0x7fadb0f55bf0 h=-&- v=-&- H:[NSView:0x7fadb30adc00]-(0)-|   (Names: '|':NSTabView:0x7fadb30adf30 )>",`
`	    "<NSLayoutConstraint:0x7fadb30b4130 H:[NSTabView:0x7fadb30adf30]-(0)-|   (Names: '|':NSView:0x7fadb30ae3d0 )>",`
`	    "<NSLayoutConstraint:0x7fadb30b4180 H:|-(0)-[NSTabView:0x7fadb30adf30]   (Names: '|':NSView:0x7fadb30ae3d0 )>",`
`	    "<NSLayoutConstraint:0x7fadb30bf4a0 H:[MyWhole360Controller:0x7fadb30c5cf0]-(NSSpace(8))-[NSBox:0x7fadb3198690'Left Stick']>",`
`	    "<NSLayoutConstraint:0x7fadb30a91c0 H:[NSBox:0x7fadb3198690'Left Stick']-(NSSpace(20))-|   (Names: '|':NSView:0x7fadb30adc00 )>",`
`	    "<NSLayoutConstraint:0x7fadb31822a0 H:[NSBox:0x7fadb3198690'Left Stick'(145)]>",`
`	    "<NSLayoutConstraint:0x7fadb30b3c80 H:[NSBox:0x7fadb30b3580'Box']-(0)-|   (Names: '|':NSView:0x7fadb30ae3d0 )>",`
`	    "<NSLayoutConstraint:0x7fadb3029940 H:|-(0)-[NSBox:0x7fadb30b3580'Box']   (Names: '|':NSView:0x7fadb30ae3d0 )>",`
`	    "<NSLayoutConstraint:0x7fadb30aff70 H:[NSBox:0x7fadb30b3580'Box'(668)]>"`
`	)`
	
`	Will attempt to recover by breaking constraint `
`	<NSLayoutConstraint:0x7fadb30c5c20 H:[MyWhole360Controller:0x7fadb30c5cf0(465)]>`
	
`	Set the NSUserDefault NSConstraintBasedLayoutVisualizeMutuallyExclusiveConstraints to YES to have -[NSWindow visualizeConstraints:] automatically called when this happens.  And/or, break on objc_exception_throw to catch this in the debugger.`
`Jan  4 00:24:25 --- last message repeated 2 times ---`
`Jan  4 00:24:25 Cerulean System Preferences[1040]: Result of kextstat = 0`
`Jan  4 00:25:37 Cerulean System Preferences[1040]: No devices found? :( device count==0, i==0`
`Jan  4 00:25:42 Cerulean kernel[0]: Google Chrome He[1049] triggered unnest of range 0x7fff92600000->0x7fff92800000 of DYLD shared region in VM map 0xc025d1856c6051c5. While not abnormal for debuggers, this increases system memory footprint until the target exits.`
 
At this point I unplugged it by mistake and plugged it back in and this message appeared:  
`Jan  4 00:26:41 Cerulean kernel[0]: start - unable to find chatpad interface`
Nothing else related afterwards.

Thanks again for being so patient and helping me out! Please let me know if I missed anything or if you need something else.

FranticRain commented:
Don't worry about the light or the cable, after seeing those pictures, those definitely aren't the problem. Unfortunately, the only important part of that log is the one that begins with `start - ` and it basically tells me that nothing went wrong. So here's another version of the driver. You'll need to check your system.log again, but this time, everything important should start with `RockCandy - `. Its important that you press a couple of buttons once the controller is plugged in. Don't go overboard though, each button press is going to put out at least 3 lines of log data. https://drive.google.com/file/d/0B5PLCKoMd4zGMC00cGNpRTNUWVU/view?usp=sharing

jebkam commented:
Just installed the new driver but still no luck detecting the input. I pasted the log below to see if by any chance there was something in there that might give a lead on what the issue is. The only unusual thing I noticed from my side was that when I connected the controller, the Mac recognized it and the light turned on but the Preference Pane was showing "No devices found". Plugging it back in fixed it, but I'm not sure if it was important to mention, since last time it got recognized immediately. Hopefully all this helps, and thanks again.

`Jan  4 14:13:49 Cerulean kernel[0]: start - unable to find chatpad interface`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - report descriptor.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - start finished.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - handleReport type: 32, size 14.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - BUTTON PACKET - 8, 9, 10, 11, 2, 3, 14, 15, 12, 13, 0, 4, 5, 6, 7`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - Orig Button: 3.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - New Buttons: 0.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - handleReport type: 32, size 14.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - BUTTON PACKET - 8, 9, 10, 11, 2, 3, 14, 15, 12, 13, 0, 4, 5, 6, 7`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - Orig Button: 3.`
`Jan  4 14:13:49 Cerulean kernel[0]: RockCandy - New Buttons: 0.`
`Jan  4 14:14:56 Cerulean System Preferences[454]: Unable to simultaneously satisfy constraints:`
`	(`
`	    "<NSLayoutConstraint:0x7fbced9b56f0 H:[MyWhole360Controller:0x7fbced9b2510(465)]>",`
`	    "<NSLayoutConstraint:0x7fbced9b5f70 H:|-(NSSpace(20))-[MyWhole360Controller:0x7fbced9b2510]   (Names: '|':NSView:0x7fbced98c510 )>",`
`	    "<NSAutoresizingMaskLayoutConstraint:0x7fbceda065c0 h=-&- v=-&- H:|-(0)-[NSView:0x7fbced98c510]   (Names: '|':NSTabView:0x7fbced8bd6a0 )>",`
`	    "<NSAutoresizingMaskLayoutConstraint:0x7fbceda06610 h=-&- v=-&- H:[NSView:0x7fbced98c510]-(0)-|   (Names: '|':NSTabView:0x7fbced8bd6a0 )>",
`	    "<NSLayoutConstraint:0x7fbced8c1bd0 H:[NSTabView:0x7fbced8bd6a0]-(0)-|   (Names: '|':NSView:0x7fbced8bdb40 )>",`
`	    "<NSLayoutConstraint:0x7fbced8c1c20 H:|-(0)-[NSTabView:0x7fbced8bd6a0]   (Names: '|':NSView:0x7fbced8bdb40 )>",`
`	    "<NSLayoutConstraint:0x7fbced9b5a70 H:[MyWhole360Controller:0x7fbced9b2510]-(NSSpace(8))-[NSBox:0x7fbced9aa230'Left Stick']>",`
`	    "<NSLayoutConstraint:0x7fbced9acfa0 H:[NSBox:0x7fbced9aa230'Left Stick']-(NSSpace(20))-|   (Names: '|':NSView:0x7fbced98c510 )>",`
`	    "<NSLayoutConstraint:0x7fbced993dc0 H:[NSBox:0x7fbced9aa230'Left Stick'(145)]>",`
`	    "<NSLayoutConstraint:0x7fbceb66c8a0 H:[NSBox:0x7fbced80dab0'Box']-(0)-|   (Names: '|':NSView:0x7fbced8bdb40 )>",`
`	    "<NSLayoutConstraint:0x7fbced8be290 H:|-(0)-[NSBox:0x7fbced80dab0'Box']   (Names: '|':NSView:0x7fbced8bdb40 )>",`
`	    "<NSLayoutConstraint:0x7fbceb6a3880 H:[NSBox:0x7fbced80dab0'Box'(668)]>"`
`	)`
	
`	Will attempt to recover by breaking constraint `
`	<NSLayoutConstraint:0x7fbced9b56f0 H:[MyWhole360Controller:0x7fbced9b2510(465)]>`
	
`	Set the NSUserDefault NSConstraintBasedLayoutVisualizeMutuallyExclusiveConstraints to YES to have -[NSWindow visualizeConstraints:] automatically called when this happens.  And/or, break on objc_exception_throw to catch this in the debugger.`
`Jan  4 14:16:05 Cerulean System Preferences[454]: No devices found? :( device count==0, i==0`

FranticRain commented:
Okay, that is super weird. You pressed two buttons, right? It looks like your controller didn't even send the button data to the driver. In fact, it's acting like you have the sync button held down, which is really weird. Can you look at the log some more and see if you can ever make `Orig Button: ` anything other than 3 by pressing buttons?

jebkam commented:
Yeah, I was started slowly expecting the log lines, but when nothing happened I started trying every button, combination of 2 buttons, 3, and so on but no response. Now even weirder, it seems to be detecting now that the Analogs are stuck in the position of the picture below. I moved the analogs around but it doesn't show any response. 
![screen shot 2016-01-04 at 3 06 26 pm](https://cloud.githubusercontent.com/assets/7463940/12099327/9d350232-b2f5-11e5-842e-6675757014d6.png)

jebkam commented:
And sorry I forgot to mention that the `Orig Button:` doesn't seem to go up from 3.

FranticRain commented:
I removed some features that might be hindering you and added a different debug output. Give it a shot. https://drive.google.com/file/d/0B5PLCKoMd4zGWlBobmhWdENXVGs/view?usp=sharing

jebkam commented:
Sorry, still no luck. Although I noticed a change in the log, it now shows this when I plug it in:

`Jan  4 16:29:22 Cerulean kernel[0]: RockCandy - Full Report: 32 0 1 14 3 0 0 984 -448 1400 -879start - unable to find chatpad interface`
`Jan  4 16:29:22 Cerulean kernel[0]: RockCandy - report descriptor.`
`Jan  4 16:29:22 Cerulean kernel[0]: RockCandy - start finished.`
`Jan  4 16:29:22 Cerulean kernel[0]: RockCandy - handleReport type: 32, size 14.`
`Jan  4 16:29:22 Cerulean kernel[0]: RockCandy - Full Report: 32 0 1 14 3 0 0 962 -424 1377 -922RockCandy - handleReport type: 32, size 14.`

Just in case I'm running on El Capitan Version 10.11.2 (15C50), and here's the controller info from the System Report:
![screen shot 2016-01-04 at 4 40 59 pm](https://cloud.githubusercontent.com/assets/7463940/12101313/f9bdfc82-b301-11e5-9a85-80b60da71b28.png)

Also I noticed a way to replicate the issue where the analogs get stuck. If I have the controller plugged in before entering the 360 Controller pane, it will show the normal controller image. Now if I have the pane open and then I plug in the controller, then it will show the analogues as in the picture I sent earlier. It resets to normal if I go back to the systems pane and then go back into 360 Controller.

FranticRain commented:
I'm going to need you to find out if it works when connected to a Windows PC or Xbox One console. It looks like its doing exactly what it should be doing from my end. It is sending perfectly valid data. And honestly, if the controller can't send data when a button is pressed, its almost assuredly not the driver's fault.

jebkam commented:
Alright then, it might take me 1 - 2 days to find one so just letting you know when to expect a response, apologies for the delay in advance. Thanks so much for helping me out, I really appreciate it! I'll keep you posted.

jebkam commented:
Ok, so I tested the controller on Windows and I couldn't get to work. It seems the controller may have been the issue all along. I apologize for the time lost, I hope all this helps someone though. I truly appreciate all your patience throughout the process.

J-Fricke commented:
@jebkam Mine also died, I wonder if they have some faulty hardware in these. When I contacted their tech support, I was told they are not intended to be used on a computer and only an xbox. I'll probably just go with Microsoft's version when I replace.
